pkg/summarizer/naive/analyzer.go

```diff
@@ -0,0 +1,187 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package naive
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	summarypb "github.com/GoogleCloudPlatform/testgrid/pb/summary"
+	"github.com/GoogleCloudPlatform/testgrid/util/types"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/sirupsen/logrus"
+	"vbom.ml/util/sortorder"
+)
+
+const analyzerName = "naiveanalyzer"
+
+var (
+	jailedRegex = regexp.MustCompile(`(JAILED )?(\/?\/?[A-Za-z0-9\/_\-.:}{]+) - \[([A-Za-z0-9\/:\-_.\s]+)\]\s*`)
+)
+
+// IntString is for sorting, primarily intended for map[string]int as implemented below
+type IntString struct {
+	s string
+	i int
+}
+
+type naiveAnalyzer struct {
+}
+
+func (na *naiveAnalyzer) GetFlakiness(results []*types.Result, minRuns int, startDate int, endDate int, tab string) *summarypb.HealthinessInfo {
+	testByEnv := make(map[string]*summarypb.TestInfo)
+	infraIssues := make(map[string]int64)
+	for _, test := range results {
+		name, env := getNameAndEnvFromTest(test.Name, tab)
+		if len(test.InfraFailures) > 0 {
+			for errorType, errorCount := range test.InfraFailures {
+				infraIssues[test.Name+"-"+errorType] += int64(errorCount)
+			}
+		}
+
+		testInfo, success := calculateNaiveFlakiness(test, minRuns)
+		if !success {
+			continue
+		}
+		testInfo.Name = name
+		testInfo.Env = env
+
+		if currTestInfo, exists := testByEnv[name]; !exists || currTestInfo.Flakiness < testInfo.Flakiness {
+			testByEnv[name] = testInfo
+		}
+	}
+	// Populate Healthiness with above calculated information
+	healthiness := createHealthiness(startDate, endDate, results, testByEnv, infraIssues)
+	return healthiness
+}
+
+func getNameAndEnvFromTest(testName string, tabName string) (string, string) {
+	name := testName
+	env := tabName
+	match := jailedRegex.FindStringSubmatch(name)
+	if match != nil {
+		name = match[2]
+		env = match[3]
+	} else {
+		logrus.Infof("Test \"%s\" could not be split into name and env, using tab name \"%s\" as its env", testName, tabName)
+	}
+	return name, env
+}
+
+func createHealthiness(startDate int, endDate int, results []*types.Result, testByEnv map[string]*summarypb.TestInfo, infraIssues map[string]int64) *summarypb.HealthinessInfo {
+	healthiness := &summarypb.HealthinessInfo{
+		StartDate:   intToTimestamp(startDate),
+		EndDate:     intToTimestamp(endDate),
+		Tests:       []*summarypb.TestInfo{},
+		InfraIssues: make(map[string]int64),
+	}
+
+	// The Go way of sorting the keys of a map in descending order
+	var keys []int
+	for k := range types.HealthyRange {
+		keys = append(keys, k)
+	}
+	sort.Sort(sort.Reverse(sort.IntSlice(keys)))
```

Comment:
if you see my comments in HealthyRange, we can just do `threadholds := []string{nonFlaky, slightlyFlaky, ...}` here

Reply:
Done